.github/workflows/build_and_run.yml

```diff
@@ -0,0 +1,30 @@
+name: Build and Test
+
+on: [push, pull_request]
```

Comment:
Your actions are running twice with this pattern. I suggest just changing it to on push.